### PR TITLE
[Material] Use primary color for selected rows and checkboxes in DataTable

### LINKED ISSUE #64314: Update DataTable to follow Material specifications
## Selected rows have incorrect color
To follow Material specifications the selected rows in a DataTable should use the primary color from the color scheme.

See: https://material.io/design/interaction/states.html#selected

## When not all rows are selected, the top checkbox should be in the "tristate"
Right now it has just a selected state regardless if all rows are selected or not.

See: https://material.io/components/data-tables

## Correct implementation
Screen Shot 2020-08-21 at 12 28 10

## Current implementation
Simulator Screen Shot - iPhone 11 Pro Max - 2020-08-21 at 12 30 09

### PR DESCRIPTION
**Pull request by perclasson**
### Light Theme
According to Material specs we should use the 8% overlay color (https://material.io/design/interaction/states.html#selected) for selected states:

### Dark Theme
In a dark theme, states should use the same overlay values as their default (or light) theme (https://material.io/design/color/dark-theme.html#states).

### Screenshots

| Light Theme  | Dark Theme |
| ------------- | ------------- |
|  |   |

## Related Issues

Closes https://github.com/flutter/flutter/issues/64314

## Tests

I added the following tests:

- DataRow renders default selected row colors
- DataRow renders checkbox with colors from Theme

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. 

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by clocksmith** [comment]
The white on green in the dark theme checkbox is hard to discern. Is there an option to change the check mark color to something darker when the background color is brighter?

**Review by clocksmith — APPROVED** [review]

**Comment by perclasson** [comment]
> The white on green in the dark theme checkbox is hard to discern. Is there an option to change the check mark color to something darker when the background color is brighter?

I agree that the white color on green is a bit hard to see, but currently there is no theme for Checkboxes: flutter.dev/go/material-theme-system-updates

File: packages/flutter/lib/src/material/data_table.dart [hunk 1]
```diff
@@ -862,13 +860,8 @@ class DataTable extends StatelessWidget {
...
     final MaterialStateProperty<Color?> defaultRowColor = MaterialStateProperty.resolveWith(
       (Set<MaterialState> states) {
-        if (states.contains(MaterialState.selected)) {
-          // The color has to be transparent so you can see the ink on
-          // the [Material].
-          // TODO(perclasson): Align with Material specs, use translucent primary color: https://github.com/flutter/flutter/issues/64314.
-          return (Theme.of(context)!.brightness == Brightness.light) ?
-            _grey100Opacity : _grey300Opacity;
-        }
+        if (states.contains(MaterialState.selected))
```
Comment (JoseAlba) [inline packages/flutter/lib/src/material/data_table.dart:863]: nit: you can change this into a ternary operator 

boolean ? exp1 : exp2

**Review by JoseAlba — APPROVED** [review]

**Review by rami-a — APPROVED** [review]
Change LGTM

**Review by clocksmith — APPROVED** [review]

**Review by HansMuller — APPROVED** [review]
LGTM

**Pull request merged**